.github/workflows/build.yml

```diff
@@ -27,24 +27,24 @@ jobs:
         os: [ubuntu-22.04, windows-2022]
         pg_major: [15, 14, 13, 12, 11, 10]
         config: [Release]
-        test_tfm: [net7.0]
+        test_tfm: [net8.0]
         include:
           - os: ubuntu-22.04
             pg_major: 15
             config: Debug
-            test_tfm: net7.0
```

Comment:
Note the netcoreapp3.1 below - we need to decide what to do about it (I think this is what causes the test failures around ImmutableDictionary)

Reply:
Funny. We have the mapping for `ImmutableDictionary<string, string?>` <=> `hstore` disabled for .NET Standard builds for no (current) reason since we had included `System.Collections.Immutable` as a dependency for it long ago.

Reply:
https://github.com/npgsql/npgsql/pull/4880 fixes this.

Reply:
I think I'll do a general review of our `#if NETXXX` directives as part of this.